### ISSUE TITLE
TypeIs does not narrow generics properly

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#8460]
```
from typing_extensions import Collection, TypeIs, Any, reveal_type


def is_int_list(x: Collection[Any]) -> TypeIs[list[int]]:
    raise NotImplementedError

def caller(x: Collection[object]) -> None:
    if is_int_list(x):
        reveal_type(x)
```

I would expect this to reveal `list[int]`, and that's what both mypy and pyanalyze produce, but pyright gives `list[object]`.

(From a Discuss post, https://discuss.python.org/t/revisiting-variance/58032/37.)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.373.
